lib/src/main/kotlin/WeatherSensorService/Internal/MockWeatherSensorReader.kt

```diff
@@ -0,0 +1,69 @@
+ //
+ // SpinDance
+ // Weather Sensor Service
+ //
+ // Copyright © 2022 SpinDance. All rights reserved.
+ //
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.SharedFlow
+import kotlinx.coroutines.flow.asSharedFlow
+import java.util.Timer
+import kotlin.concurrent.*
+import kotlin.random.Random
+import java.time.LocalDateTime
+
+private class Constants {
+    companion object {
+        var READER_INTERVAL: UInt = 1U
+        val TEMPERATURE_RANGE : ClosedRange<Double> = -40.0..40.0
+        val HUMIDITY_RANGE : ClosedRange<Double> = 0.0..100.0
+        val PRESSURE_RANGE : ClosedRange<Double> = 95.0..105.0
+    }
+}
+
+class MockWeatherSensorReader: WeatherSensorReaderType {
+     
+    private var timer: Timer? = null
+
+    override var readerInterval : UInt = Constants.READER_INTERVAL
+        
+    
+    private val flow: MutableSharedFlow<WeatherSensorReading> = MutableSharedFlow<WeatherSensorReading>(replay = 1)
```

Comment:
I can't remember, what is the replay for?

Reply:
It specifies the number of values that it will keep in cache. From [documentation](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.flow/-shared-flow/): "A shared flow keeps a specific number of the most recent values in its replay cache. Every new subscriber first gets the values from the replay cache and then gets new emitted values. The maximum size of the replay cache is specified when the shared flow is created by the replay parameter."